broker/src/main/java/org/apache/rocketmq/broker/longpolling/PullRequestHoldService.java

```diff
@@ -54,6 +56,11 @@ public void suspendPullRequest(final String topic, final int queueId, final Pull
 
         pullRequest.getRequestCommand().setSuspended(true);
         mpr.addPullRequest(pullRequest);
+        // check the last notify maxOffset
```

Comment:
if the pull request been put in the `pullRequestTable`, will it be processed repeated?

Reply:
Won't. The implementation of notifyMessageArriving guarantees that each request will only be processed once even in concurrent scenarios.